plugin/trino-teradata/src/main/java/io/trino/plugin/teradata/TeradataClient.java

```diff
@@ -0,0 +1,1431 @@
+        // 1. NULL-only filters are always safe
+        if (domain.isOnlyNull()) {
+            return FULL_PUSHDOWN.apply(session, domain);
+        }
+
+        Domain simplifiedDomain = domain.simplify(getDomainCompactionThreshold(session));
+        if (!simplifiedDomain.getValues().isDiscreteSet()) {
+            // Push down inequality predicate
+            ValueSet complement = simplifiedDomain.getValues().complement();
+            if (complement.isDiscreteSet()) {
+                return FULL_PUSHDOWN.apply(session, simplifiedDomain);
+            }
+            // Domain#simplify can turn a discrete set into a range predicate
+            // Push down of range predicate for varchar/char types could lead to incorrect results
+            // when the remote database is case-insensitive
+            return DISABLE_PUSHDOWN.apply(session, domain);
+        }
+        return FULL_PUSHDOWN.apply(session, simplifiedDomain);
+    };
+    /**
+     * Maximum fallback number of distinct values (NDV) for statistics estimation.
+     */
+    private static final long MAX_FALLBACK_NDV = 1_000_000L;
+    /**
+     * Default fallback fraction for NDV estimation (10% of row count).
+     */
+    private static final double DEFAULT_FALLBACK_FRACTION = 0.1;
+    /**
+     * Maximum supported timestamp precision in Teradata.
+     */
+    private static final int TERADATA_MAX_SUPPORTED_TIMESTAMP_PRECISION = 6;
+    /**
+     * Trino type representing JSON columns.
+     */
+    private final Type jsonType;
+    /**
+     * Teradata String case sensitivity mode.
+     */
+    private final TeradataConfig.TeradataCaseSensitivity teradataJDBCCaseSensitivity;
+    /**
+     * Flag indicating if statistics collection is enabled.
+     */
+    private final boolean statisticsEnabled;
+    /**
+     * Expression rewriter for translating connector expressions to Teradata SQL.
+     */
+    private ConnectorExpressionRewriter<ParameterizedExpression> connectorExpressionRewriter;
+    /**
+     * Aggregate function rewriter for translating Trino aggregates to Teradata SQL.
+     */
+    private AggregateFunctionRewriter<JdbcExpression, ?> aggregateFunctionRewriter;
+
+    /**
+     * Constructs a new TeradataClient instance.
+     *
+     * @param config base JDBC configuration
+     * @param teradataConfig Teradata-specific configuration
+     * @param connectionFactory factory to create JDBC connections
+     * @param queryBuilder query builder for SQL queries
+     * @param identifierMapping mapping for identifiers such as column names
+     * @param remoteQueryModifier optional modifier for remote queries
+     */
+    @Inject
+    public TeradataClient(BaseJdbcConfig config, TeradataConfig teradataConfig, JdbcStatisticsConfig statisticsConfig, ConnectionFactory connectionFactory, QueryBuilder queryBuilder, TypeManager typeManager, IdentifierMapping identifierMapping, RemoteQueryModifier remoteQueryModifier)
```

Comment:
Put parameters on separate lines.

Reply:
addressed in new PR https://github.com/trinodb/trino/pull/26731